numerics/src/tools/NumericsSparseMatrix.c

```diff
@@ -44,10 +45,20 @@ typedef struct
   size_t indx;
 } sort_indices_struct;
 
+#ifdef HAVE_SORT
 #define SORT_NAME sorter
 #define SORT_TYPE sort_indices_struct
 #define SORT_CMP(x, y) ((x).i - (y).i)
 #include "sort.h"
+#else
+#include "stdlib.h" // qsort
+static int sort_indices_struct_cmp(const void *a, const void *b)
+{
+  const sort_indices_struct *sa = (const sort_indices_struct *) a;
+  const sort_indices_struct *sb = (const sort_indices_struct *) b;
+  return (sa->i > sb->i) - (sa->i < sb->i);
+}
```

Comment:
I was just curious about that magic: is sa->i - sb->i not sufficient? If my C99 is not too rusty, sa->i > sb->i evaluates to a boolean. So the result is a difference of boolean promoted to int. It is a bit confusing for me. I have to admit that I don't remember that piece of code very well ...